### PR TITLE
tools/c7n_policystream - bump pygit2 dependency

### PR DESCRIPTION
**Pull request by ajkerrigan**
Let poetry update pygit2 beyond 1.9. Today that means locking pygit `1.10.1`, because `1.11+` requires Python 3.8 and we still support 3.7.

Notably this pulls in prebuilt wheels for Python 3.11 (see also #8057 )

### CODE REVIEW

File: tools/c7n_policystream/pyproject.toml [hunk 1]
```diff
@@ -23,7 +23,7 @@ python = "^3.7"
 click = "^8.0"
 requests = "^2.22.0"
 pyyaml = ">=5.4.0"
-pygit2 = "~1.9"
+pygit2 = "^1.9"
```
Comment (thisisshi) [inline tools/c7n_policystream/pyproject.toml:26]: shouldnt this be `~1.10` then if we dont want it to go to `1.11` yet?

Comment (kapilt) [reply tools/c7n_policystream/pyproject.toml:26]: We're already somewhat differential in the stack tfparse for c7n-left is 3.10+ only per ci, as we look to 3.11, and various blockers I think it's realistic we may have to fuzz the edges, as long as we note the matrix on readme and default to batteries included on docker I think we can hit the best of both worlds, I hope.

**Review by kapilt — APPROVED** [review]
lgtm

File: tools/c7n_policystream/pyproject.toml (hunk 1 above)
Comment (kapilt) [reply tools/c7n_policystream/pyproject.toml:26]: its a reasonable consideration for us to consider dropping 3.7 at this point, its end of life upstream in a few months.

**Review by thisisshi — APPROVED** [review]

**Pull request merged**